homeassistant/components/fjaraskupan/sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Support for sensors."""
+from __future__ import annotations
+
+from fjaraskupan import Device, State
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_SIGNAL_STRENGTH,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import SIGNAL_STRENGTH_DECIBELS_MILLIWATT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from . import DeviceState, async_setup_entry_platform
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors dynamically through discovery."""
+
+    def _constructor(device_state: DeviceState):
```

Comment:
Would be good to type the return value here too.

Reply:
Why? it is deduced from the code, and the call to async_setup_entry_platform is typed. So if an error in return value was made, it would not be accepted by mypy.

Reply:
We always type the signature completely when adding type annotations.